static/app/utils/getPreloadedData.ts

```diff
@@ -4,8 +4,21 @@ export async function getPreloadedDataPromise(
   fallback: () => Promise<any>,
   usePreload?: boolean
 ) {
+  const data = (window as any).__sentry_preload;
+  /**
+   * Save the fallback promise to `__sentry_preload` to allow the sudo modal to wait
+   * for the promise to resolve
+   */
+  const wrappedFallback = () => {
```

Comment:
any chance that has file is tested and we could add a unit test to make sure the wrappedFallback is invoked as expected?